yb-voyager/src/srcdb/mysql.go

```diff
@@ -192,3 +193,26 @@ func (ms *MySQL) GetCharset() (string, error) {
 func (ms *MySQL) FilterUnsupportedTables(tableList []*sqlname.SourceName) []*sqlname.SourceName {
 	return tableList
 }
+
+func (ms *MySQL) FilterEmptyTables(tableList []*sqlname.SourceName) []*sqlname.SourceName {
+	nonEmptyTableList := make([]*sqlname.SourceName, 0)
+	for _, tableName := range tableList {
+		query := fmt.Sprintf(`SELECT false FROM %s.%s LIMIT 1;`, tableName.SchemaName.Unquoted, tableName.ObjectName.MinQuoted)
+		fmt.Printf("query: %v\n", query)
+		var empty bool
+		err := ms.db.QueryRow(query).Scan(&empty)
+		if err != nil {
+			if err == pgx.ErrNoRows {
+				empty = true
+			} else {
+				utils.ErrExit("error in querying table %v: %v", tableName, err)
+			}
+		}
+		if !empty {
+			nonEmptyTableList = append(nonEmptyTableList, tableName)
+		} else {
+			log.Infof("Skipping empty table %v", tableName)
+		}
+	}
+	return nonEmptyTableList
```

Comment:
There is a lot of code duplication in these implementations. Refactor it to have code like:
```
query := fmt.Sprintf(`SELECT false FROM %s.%s LIMIT 1;`, tableName.SchemaName.Unquoted, tableName.ObjectName.MinQuoted)
if isEmpty(ms.db, query) {
  nonEmptyTableList = append(nonEmptyTableList, tableName)
}
```
The implementation of the `isEmpty(*sql.DB, query)` must be database independent.